woocommerce/google-listings-and-ads
# Fetch assets from other Asset Groups using the final url

### Changes proposed in this Pull Request:



Closes part of https://github.com/woocommerce/google-listings-and-ads/issues/1780

This PR allows populating assets (images, descriptions, headlines..) from other asset groups using the same final URL than the URL from a specific product, page, post or category. In case there are no assets available in other asset groups it will fetch the WP Assets.

See the following PRs for more context: 

- https://github.com/woocommerce/google-listings-and-ads/pull/1795
- https://github.com/woocommerce/google-listings-and-ads/pull/1797

### Detailed test instructions:


1. Look for a post, page, product or category and get the ID. 
3. Make a request to `gla/assets/suggestions?type=post|term&id=POST_ID|TERM_ID`. If it is category or taxonomy the `type` is `term` otherwise it will be `post`
4. If there are not asset groups with the post/term URL, the response should include the assets included in the WP post or term.
5. If there was an asset group with the same `final_url` (see `final_url` from the previous response) it should show the assets from that asset group.
6. If you do not have any asset group using that `final_url`, you can go to [Google Ads ](https://ads.google.com/intl/en_EN/home/), choose a Campaign -> Asset Groups -> Edit Asset Group -> Update the final URL with the post or term URL that you got earlier.

![image](https://user-images.githubusercontent.com/2488994/210385047-544d373c-82b1-4e93-8a98-4d90ccec4c10.png)

7. Repeat the steps and you should get the Assets from the Asset group.



### Additional details:
- If there is more than one asset group using the same final URL, we would suggest the assets found in the first asset group returned from the Ads API, the reason for this is that we do not want to exceed the maximum allowed assets and to avoid repeated assets. See here all the assets requirements: https://developers.google.com/google-ads/api/docs/performance-max/assets.
- If you are using a local URL and you would like to change it temporarily so it can match with the URL in the asset group you can use the following filter:

```
add_filter('home_url', function ($url) {
   return  str_replace('http://localhost', 'https://test.ngrok.io', $url);
}, 10, 1);
```



### Changelog entry
